src/control/server/storage_nvme.go

```diff
@@ -500,31 +499,31 @@ func loadNamespaces(ctrlrPciAddr string, nss []spdk.Namespace) (_nss types.NvmeN
 	return
 }
 
-// loadHealthStats converts a slice of DeviceHealth into protobuf equivalent.
-// Implemented as a pure function.
-func loadHealthStats(health []spdk.DeviceHealth) (_health types.NvmeHealthstats) {
-	for _, h := range health {
-		_health = append(
-			_health,
-			&ctlpb.NvmeController_Health{
-				Temp:            h.Temp,
-				Tempwarn:        h.TempWarnTime,
-				Tempcrit:        h.TempCritTime,
-				Ctrlbusy:        h.CtrlBusyTime,
-				Powercycles:     h.PowerCycles,
-				Poweronhours:    h.PowerOnHours,
-				Unsafeshutdowns: h.UnsafeShutdowns,
-				Mediaerrors:     h.MediaErrors,
-				Errorlogs:       h.ErrorLogEntries,
-				Tempwarning:     h.TempWarn,
-				Availspare:      h.AvailSpareWarn,
-				Reliability:     h.ReliabilityWarn,
-				Readonly:        h.ReadOnlyWarn,
-				Volatilemem:     h.VolatileWarn,
-			})
+// loadHealthStats find health statistics for a given control identified by PCI
+// address.
+func loadHealthStats(ctrlrPciAddr string, hss []spdk.DeviceHealth) *ctlpb.NvmeController_Health {
+	for _, hs := range hss {
```

Comment:
This has O(n) complexity. Probably not a huge deal at the scale of a list of devices on a single node, but if you changed your function which generates the slice to instead generate a map with the pci address as the key, then you'd get O(1) without much more work.

Reply:
cool, will fix in the next PR (have a follow-on for this area)